core/src/main/java/cz/o2/proxima/scheme/SchemaDescriptors.java

```diff
@@ -0,0 +1,885 @@
+   *
+   * @param name structure name
+   * @param fields list of fields in structure
+   * @param defaultValueFactory default value factory
+   * @param <T> structure type
+   * @return Structure type descriptor
+   */
+  public static <T> StructureTypeDescriptor<T> structures(
+      String name, List<FieldTypeDescriptor<?>> fields, Factory<T> defaultValueFactory) {
+    return new StructureTypeDescriptor<>(name, fields, defaultValueFactory);
+  }
+
+  /**
+   * Create {@link FieldTypeDescriptor} of {@link PrimitiveTypeDescriptor} with name.
+   *
+   * @param name name of field
+   * @param descriptor value descriptor
+   * @param <T> value type
+   * @return field descriptor
+   */
+  public static <T> FieldTypeDescriptor<T> fields(
+      String name, PrimitiveTypeDescriptor<T> descriptor) {
+    return fields(name, descriptor.toTypeDescriptor());
+  }
+
+  /**
+   * Create {@link FieldTypeDescriptor} of {@link ArrayTypeDescriptor} with name.
+   *
+   * @param name name of field
+   * @param descriptor value descriptor
+   * @param <T> value type
+   * @return field descriptor
+   */
+  public static <T> FieldTypeDescriptor<T> fields(String name, ArrayTypeDescriptor<T> descriptor) {
+    return fields(name, descriptor.toTypeDescriptor());
+  }
+
+  /**
+   * Create {@link FieldTypeDescriptor} of {@link StructureTypeDescriptor} with name
+   *
+   * @param name name of field
+   * @param descriptor value descriptor
+   * @param <T> value type
+   * @return field descriptor
+   */
+  public static <T> FieldTypeDescriptor<T> fields(
+      String name, StructureTypeDescriptor<T> descriptor) {
+    return fields(name, descriptor.toTypeDescriptor());
+  }
+
+  /**
+   * Create {@link FieldTypeDescriptor} of {@link SchemaTypeDescriptor} with name
+   *
+   * @param name name of field
+   * @param descriptor value descriptor
+   * @param <T> value type
+   * @return field descriptor
+   */
+  public static <T> FieldTypeDescriptor<T> fields(String name, SchemaTypeDescriptor<T> descriptor) {
+    return fields(name, descriptor, null);
+  }
+
+  /**
+   * Create {@link FieldTypeDescriptor} of {@link SchemaTypeDescriptor} with name and default value.
+   *
+   * @param name name of field
+   * @param descriptor value descriptor
+   * @param defaultValueFactory default value factory
+   * @param <T> value type
+   * @return field descriptor
+   */
+  public static <T> FieldTypeDescriptor<T> fields(
+      String name, SchemaTypeDescriptor<T> descriptor, Factory<T> defaultValueFactory) {
+    return new FieldTypeDescriptor<>(name, descriptor, defaultValueFactory);
+  }
+
+  /**
+   * Generic type descriptor. Parent class for other types
+   *
+   * @param <T> value type
+   */
+  public abstract static class GenericTypeDescriptor<T> implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    /** Value type */
+    @Getter protected final AttributeValueType type;
```

Comment:
This makes no sense for `StructureTypeDescriptor`. We probably should move that to some other superclass that could be used for descriptors of single type.

Reply:
`StructureTypeDescriptor` still have type `STRUCTURE` and I think It still makes sense as you can have structure with field which is another structure etc. Don't see any big advantage of having new superclass

Reply:
Ah, right.